src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs

```diff
@@ -167,6 +168,22 @@ public async Task SslStream_NetworkStream_Renegotiation_Succeeds()
             }
         }
 
+        [Fact]
+        public async Task SslStream_NestedAuth_Throws()
+        {
+            VirtualNetwork network = new VirtualNetwork();
+
+            using (var clientStream = new VirtualNetworkStream(network, isServer: false))
+            using (var serverStream = new VirtualNetworkStream(network, isServer: true))
+            using (var ssl = new SslStream(clientStream))
+            {
+                // Start handshake.
+                Task task = ssl.AuthenticateAsClientAsync("foo.com", null, SslProtocols.Tls12, false);
+                // Do it again without waiting for previous one to finish.
+                await Assert.ThrowsAnyAsync<InvalidOperationException>(() => ssl.AuthenticateAsClientAsync("foo.com", null, SslProtocols.Tls12, false));
```

Comment:
Why is this .ThrowsAnyAsync and not .ThrowAsync? Is there some exception derived from InvalidOperationException that is thrown?

Reply:
fixed. This is oversight when I copy base structure from another test,